editor/src/components/editor/actions/actions.ts

```diff
@@ -4197,6 +4201,26 @@ export const UPDATE_FNS = {
       },
     }
   },
+  SET_SCROLL_ANIMATION: (
+    action: SetScrollAnimation,
+    editor: EditorModel,
+    dispatch: EditorDispatch,
+  ): EditorModel => {
+    if (action.value) {
+      canvasScrollAnimationTimer = window.setTimeout(() => {
```

Comment:
it looks like you are missing a line that checks if `canvasScrollAnimationTimer != null` then clearTimeout:

```suggestion
      if (canvasScrollAnimationTimer != null) {
        clearTimeout(canvasScrollAnimationTimer)
      }
      canvasScrollAnimationTimer = window.setTimeout(() => {
```

because if multiple `setScrollAnimation(true)` are dispatched in short order, I think we want to extend the time the value stays true.

Reply:
fixed